### DISCUSSION TITLE
Mace in lammps plus D3

### DISCUSSION THREAD

**Discussion by Tingliangstu** [ACEsuit/mace#702] [Q&A] [answered]
Hi all,

Can D3 correction be added when running mace or MACE-MP-0 in lammps? How can this be achieved?

I don't want to run MD in ASE because the efficiency is too slow and there are some problems with its ensemble.

Thank you very much.

Best,
LIANG Ting

**Comment by ilyes319** [comment]
Dear @Tingliangstu,

While it is theoritically possible to compile the D3 correction with jit and then load it in LAMMPS, there is no easy way to do that at the moment, unfortunately. We aim to release models trained with the correction in the coming months.

**Reply by Tingliangstu** [reply]
So to use MACE-MP-0+D3, one can only do it in ASE calculator, right?

Thanks so much！

**Comment by gabor1** [comment]
you can use LAMMPS overlay, and add on a d3 that way, but you'll lose some of the lammps efficiency.

**Reply by Tingliangstu** [reply]
thanks so much for your reply. but can u give some example for quickly add D3,

```
pair_style 	 mace no_domain_decomposition
pair_coeff 	 * * ../medium-2023-12-03-mace-128-L1_epoch-199.model-lammps.pt C
```

How can I add D3 this way? Can you give me a clear example?

Thank you very much for your help.

**Comment by gabor1** [comment]
I wonder how these people do it. 

https://nanolabo-doc.readthedocs.io/en/v2.8/usage/m3gnet.html

**Reply by gabor1** [reply]
@vue1999 can you try and look into this?

**Reply by Tingliangstu** [reply]
Thanks so much for your reply.

For m3gnet and chgnet, they used the calculator interface of python to connect to lammps, which I'm afraid is different from mace. They can add D3 through the package dftd3. Can MACE do this?

Best,
Ting

**Comment by gabor1** [comment, marked as answer]
In ase we use dftd3 as well (and there is a fork of it that uses skin distance so you don't need to recompute the neighbor list every time)

**Comment by gabor1** [comment]
https://github.com/CheukHinHoJerry/torch-dftd
